nixos/modules/services/matrix/matrix-sliding-sync.nix

```diff
@@ -80,8 +80,11 @@ in
       } ];
     };
 
-    systemd.services.matrix-sliding-sync = {
-      after = lib.optional cfg.createDatabase "postgresql.service";
+    systemd.services.matrix-sliding-sync = rec {
+      after =
+        lib.optional cfg.createDatabase "postgresql.service"
+        ++ lib.optional config.services.matrix-synapse.enable "matrix-synapse.service";
+      wants = after;
```

Comment:
Don't you need `requires` to propagate a restart (that's how I understand your PR describe at least):

> [...] configures matrix-sliding-sync to automatically restart as the service fails if synapse is not reachable yet.

Also, did the service actually break? I cannot reproduce this, here are the logs from `matrix-sliding-sync.service` when doing a `systemctl restart matrix-synapse.target`:

```
Nov 04 13:44:50 soost syncv3[2172]: 13:44:50 WRN Poller: sync v2 poll returned temporary error error="DoSyncV2: response returned 502 Bad Gateway" code=502 device_id=YDVFFUECYG user_id=@ma27:nicht-so.sexy
Nov 04 13:44:50 soost syncv3[2172]: 13:44:50 WRN Poller: waiting before next poll device_id=YDVFFUECYG duration=3s fail-count=1 user_id=@ma27:nicht-so.sexy
```

In other words I'm not sure if this would be desirable.

Reply:
The automatic restart is meant to describe the `Restart` and `RestartSec` options set in the service.
I don't have logs right now but I did have sporadic issues with `matrix-sliding-sync.service` failing to start because it couldn't reach the upstream homeserver. On deployments where both the sync proxy and the homeserver are running on the same host, `Wants` and `After` make sure the sync proxy isn't started without the homeserver (though we could change this to `Requires` if we want to). On deployments where sync proxy and homeserver are running on different hosts, the `Restart` and `RestartSec` make sure the sync proxy is restarted automatically when it fails.

Reply:
> The automatic restart is meant to describe the Restart and RestartSec options set in the service

OK never mind then, I misunderstood you then, I thought this was about actions to the synapse service being propagated to sliding sync. Fwiw, I agree about after/wants here.

> but I did have sporadic issues with matrix-sliding-sync.service failing to start because it couldn't reach the upstream homeserver

The example above is about a restart only, I haven't checked closely how it behaves on startup.

Nothing else to discuss here IMHO, so I'll resolve.